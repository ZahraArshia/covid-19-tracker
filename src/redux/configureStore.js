import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import countriesReducer from './modules/country';

const reducer = combineReducers({
  countriesReducer,
});

const configureStore = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default configureStore;
