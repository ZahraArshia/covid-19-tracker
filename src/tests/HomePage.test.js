/* eslint-disable camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from './mocks/configureStore-mock';
import Countries, { Country } from '../components/HomePage';

describe('Home page test', () => {
  const id = 'iran';
  const name = 'Iran';
  const today_confirmed = 100;
  test('country cards test', () => {
    const tree = renderer
      .create(
        <Provider store={configureStore}>
          <BrowserRouter>
            <Country id={id} name={name} today_confirmed={today_confirmed} />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('countries container test', () => {
    const tree = renderer
      .create(
        <Provider store={configureStore}>
          <BrowserRouter>
            <Countries />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
