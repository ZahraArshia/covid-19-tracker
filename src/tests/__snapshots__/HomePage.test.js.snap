// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test countries container test 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <img
          alt="virus-transmission"
          src="https://img.icons8.com/external-wanicon-lineal-wanicon/120/ffffff/external-world-virus-transmission-wanicon-lineal-wanicon.png"
        />
        <div>
          <h1>
            2022
          </h1>
          <h3>
            covid-19 world tracker
          </h3>
        </div>
      </div>
    </div>
  </header>
  <div
    className="searchBar"
  >
    <div
      className="dateSelection"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-21
      </h3>
    </div>
    <input
      className="searchField"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Search Countries"
      type="text"
      value=""
    />
  </div>
  <ul
    className="Countriescontainer"
  />
</div>
`;

exports[`Home page test country cards test 1`] = `
<div
  className="card"
>
  <a
    className=""
    href="/Iran"
    onClick={[Function]}
  >
    <li
      className="countryCard"
    >
      <div
        className="mapBox"
      >
        <img
          alt="map"
          className="map"
          onError={[Function]}
          src="https://mapsvg.com/static/maps/geo-calibrated/iran.svg"
        />
      </div>
      <h4>
        Iran
      </h4>
      <h4>
        100
      </h4>
    </li>
  </a>
</div>
`;
